shelley/chain-and-ledger/executable-spec/test/Examples.hs

```diff
@@ -96,47 +104,33 @@ mkAddr :: (KeyPair, KeyPair) -> Addr
 mkAddr (payKey, stakeKey) =
   AddrBase (KeyHashObj . hashKey $ vKey payKey) (KeyHashObj . hashKey $ vKey stakeKey)
 
-gerolamoVKG :: VKeyGenesis
-gerolamoVKG = VKeyGenesis 1501 :: VKeyGenesis
-
-gerolamoCold :: KeyPair
-gerolamoCold = KeyPair vk sk
-  where (sk, vk) = mkKeyPair (1501, 0, 0, 0, 0)
-
-gerolamoVRF :: KeyPair
-gerolamoVRF = KeyPair vk sk
-  where (sk, vk) = mkKeyPair (1501, 0, 0, 0, 1)
-
-gerolamoHot :: (SKeyES, VKeyES)
-gerolamoHot = mkKESKeyPair (0, 0, 0, 0, 0)
+data AllPoolKeys = AllPoolKeys
+  { cold :: KeyPair
+  , vrf :: KeyPair
+  , hot :: (SKeyES, VKeyES)
+  , hk  :: KeyHash
+  } deriving (Show, Eq)
 
-lodovicoVKG :: VKeyGenesis
-lodovicoVKG = VKeyGenesis 1521 :: VKeyGenesis
+mkAllPoolKeys :: Word64 -> AllPoolKeys
+mkAllPoolKeys w = AllPoolKeys (KeyPair vkCold skCold)
+                              (KeyPair vkVrf skVrf)
+                              (mkKESKeyPair (w, 0, 0, 0, 3))
+                              (hashKey vkCold)
+  where
+    (skCold, vkCold) = mkKeyPair (w, 0, 0, 0, 1)
+    (skVrf, vkVrf) = mkKeyPair (w, 0, 0, 0, 2)
 
-lodovicoCold :: KeyPair
-lodovicoCold = KeyPair vk sk
-  where (sk, vk) = mkKeyPair (1521, 0, 0, 0, 0)
+coreNodes :: [(VKeyGenesis, AllPoolKeys)]
+coreNodes = [(mkVKGen (x, 0, 0, 0, 0), mkAllPoolKeys x) | x <-[101..107]]
 
-lodovicoVRF :: KeyPair
-lodovicoVRF = KeyPair vk sk
-  where (sk, vk) = mkKeyPair (1521, 0, 0, 0, 1)
+coreNodeVKG :: Int -> VKeyGenesis
+coreNodeVKG = fst . (coreNodes !!)
 
-lodovicoHot :: (SKeyES, VKeyES)
-lodovicoHot = mkKESKeyPair (1, 0, 0, 0, 0)
+coreNodeKeys :: Int -> AllPoolKeys
+coreNodeKeys = snd . (coreNodes !!)
 
-nicoloVKG :: VKeyGenesis
-nicoloVKG = VKeyGenesis 1499 :: VKeyGenesis
-
-nicoloCold :: KeyPair
-nicoloCold = KeyPair vk sk
-  where (sk, vk) = mkKeyPair (1499, 0, 0, 0, 0)
-
-nicoloVRF :: KeyPair
-nicoloVRF = KeyPair vk sk
-  where (sk, vk) = mkKeyPair (1499, 0, 0, 0, 1)
-
-nicoloHot :: (SKeyES, VKeyES)
-nicoloHot = mkKESKeyPair (1, 0, 0, 0, 0)
+dms :: Map VKeyGenesis VKey
+dms = Map.fromList [ (coreNodeVKG n, vKey $ cold $ coreNodeKeys n) | n <- [0..6]]
```

Comment:
do those definitions depend on the number of core nodes? Maybe extract the list `coreNodes = [0..6]` as a constant and use it here and above something like `[node + 101 | node <- coreNodes]` ?

Reply:
yea, it would be better to have only one place where the number of nodes can has to be changed. actually we can generate the `dms` mapping directly from the `coreNodes` list above, but I still like the idea of a constant for the number of nodes. thanks!